<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>index – Luca Scrucca</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../media/favicon.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-16f0be44cf0b9c8dc4714857273be9e1.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark-c76bc2edb0e532e390ad582fcbf98d52.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-71d68a64f28ab18b51d4c26199cd7cdf.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark-69a26e936a92c2b36fd0ad26839128b3.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

<script type="text/javascript">

(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', 'UA-5097379-1', 'auto');

ga('send', {
  hitType: 'pageview',
  'anonymizeIp': true,
});
</script>


</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Luca Scrucca</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html#research"> 
<span class="menu-text">Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../index.html#software"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../index.html#teaching"> 
<span class="menu-text">Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../index.html#contact"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#books" id="toc-books" class="nav-link active" data-scroll-target="#books">Books</a></li>
  <li><a href="#journal-articles" id="toc-journal-articles" class="nav-link" data-scroll-target="#journal-articles">Journal articles</a></li>
  <li><a href="#book-chapters" id="toc-book-chapters" class="nav-link" data-scroll-target="#book-chapters">Book chapters</a></li>
  <li><a href="#conferences-and-workshops" id="toc-conferences-and-workshops" class="nav-link" data-scroll-target="#conferences-and-workshops">Conferences and workshops</a></li>
  <li><a href="#reports" id="toc-reports" class="nav-link" data-scroll-target="#reports">Reports</a></li>
  <li><a href="#miscellanea" id="toc-miscellanea" class="nav-link" data-scroll-target="#miscellanea">Miscellanea</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block"></header>




<section id="books" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="books">Books</h2>
<div class="bibitem">
<p>Scrucca L., Fraley C., Murphy T. B. and Raftery A. E. (2023) <strong>Model-Based Clustering, Classification, and Density Estimation Using mclust in R</strong>, Chapman &amp; Hall/CRC, ISBN: 978-1032234953. <a href="https://doi.org/10.1201/9781003277965" class="uri">https://doi.org/10.1201/9781003277965</a></p>
</div>
</section>
<section id="journal-articles" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="journal-articles">Journal articles</h2>
<div class="bibitem">
<p>Scrucca L., Karlis D. (2025) A model-based approach to shot charts estimation in basketball. <strong>Computational Statistics</strong>. <a href="https://doi.org/10.1007/s00180-025-01599-1" class="uri">https://doi.org/10.1007/s00180-025-01599-1</a></p>
<p>Scrucca L., (2024) A model-based clustering approach for bounded data using transformation-based Gaussian mixture models. <strong>Under review</strong>.<br>
arXiv pre-print available at <a href="http://arxiv.org/abs/2412.13572" class="uri">http://arxiv.org/abs/2412.13572</a></p>
<p>Tancini D., Scrucca L., Bartolucci F. (2024) Unimodal density-based clustering and merging algorithm using Gaussian mixtures. <strong>Under review</strong>.</p>
<p>Scrucca L., (2024) Assessing uncertainty in Gaussian mixtures-based entropy estimation. <strong>Under review</strong>.<br>
arXiv pre-print available at <a href="https://arxiv.org/abs/2405.17265" class="uri">https://arxiv.org/abs/2405.17265</a></p>
<p>Turzo T., Chirieleison C., Montrone A., Scrucca L. (2024) Sustainability standards and decertification: a Bayesian analysis of SA800. <strong>Under review</strong>.</p>
<p>Scrucca L. (2024) Entropy-based volatility analysis of financial log-returns using Gaussian mixture models. <strong>Entropy</strong>, 26(11), 907. <a href="https://doi.org/10.3390/e26110907" class="uri">https://doi.org/10.3390/e26110907</a></p>
<p>Tiburtini M., Scrucca L. and Peruzzi L. (2024) A guide to plant morphometrics using Gaussian Mixture Models. <strong>Under review</strong>.<br>
bioRxiv pre-print available at <a href="https://doi.org/10.1101/2024.04.21.590472" class="uri">https://doi.org/10.1101/2024.04.21.590472</a></p>
<p>Scrucca L. (2023) On the influence of data imbalance on supervised Gaussian mixture models. <strong>Algorithms</strong>, 16(12), 563. <a href="https://doi.org/10.3390/a16120563" class="uri">https://doi.org/10.3390/a16120563</a></p>
<p>Scrucca L. (2023) Entropy-based anomaly detection for Gaussian mixture modeling. <strong>Algorithms</strong>, 16(4), 195. <a href="https://doi.org/10.3390/a16040195" class="uri">https://doi.org/10.3390/a16040195</a></p>
<p>Robin S., Scrucca L. (2023) Mixture-based estimation of entropy. <strong>Computational Statistics &amp; Data Analysis</strong>, 177, 107582. <a href="https://doi.org/10.1016/j.csda.2022.107582" class="uri">https://doi.org/10.1016/j.csda.2022.107582</a></p>
<p>Santini F., Elisei L., Malmi T. and Scrucca L. (2022) Management-control-system configurations in medium-sized mechanical-engineering firms: an exploratory analysis, <strong>Accounting Research Journal</strong>, 35(6), 834–853. <a href="https://doi.org/10.1108/ARJ-06-2021-0168" class="uri">https://doi.org/10.1108/ARJ-06-2021-0168</a></p>
<p>Scrucca L. (2022) Modal clustering on PPGMMGA projection subspace. <strong>Australian &amp; New Zealand Journal of Statistics</strong>, 64:2, 158–170. <a href="https://doi.org/10.1111/anzs.12360" class="uri">https://doi.org/10.1111/anzs.12360</a></p>
<p>Scrucca L. (2022) A COVINDEX based on a GAM beta regression model with an application to the COVID-19 pandemic in Italy. <strong>Statistical Methods &amp; Applications</strong>, 31:4, 881–900. <a href="https://doi.org/10.1007/s10260-021-00617-y" class="uri">https://doi.org/10.1007/s10260-021-00617-y</a></p>
<p>Chirieleison C., Montrone A., Scrucca L. (2022) Destination labels for historic villages: the impact on perception, experience, and satisfaction. <strong>Tourism and Hospitality Research</strong>, 22:2, 164–179. <a href="https://doi.org/10.1177/14673584211020788" class="uri">https://doi.org/10.1177/14673584211020788</a></p>
<p>Scrucca L. (2021) A fast and efficient Modal EM algorithm for Gaussian mixtures. <strong>Statistical Analysis and Data Mining</strong>, 14:4, 305–314. <a href="https://doi.org/10.1002/sam.11527" class="uri">https://doi.org/10.1002/sam.11527</a></p>
<p>Casa A., Scrucca L., Menardi G. (2021) Better than the best? Answers via model ensemble in density-based clustering. <strong>Advances in Data Analysis and Classification</strong>, 15, 599—623. <a href="https://doi.org/10.1007/s11634-020-00423-6" class="uri">https://doi.org/10.1007/s11634-020-00423-6</a></p>
<p>Chirieleison C., Montrone A., Scrucca L. (2020) Event sustainability and sustainable transportation: a positive reciprocal influence. <strong>Journal of Sustainable Tourism</strong>, 28:2, 240–262. <a href="https://doi.org/10.1080/09669582.2019.1607361" class="uri">https://doi.org/10.1080/09669582.2019.1607361</a></p>
<p>Scrucca L., Serafini A. (2019) Projection pursuit based on Gaussian mixtures and evolutionary algorithms. <strong>Journal of Computational and Graphical Statistics</strong>, 28:4, 847–860. <a href="https://doi.org/10.1080/10618600.2019.1598871" class="uri">https://doi.org/10.1080/10618600.2019.1598871</a></p>
<p>O’Hagan A., Murphy T. B., Scrucca L. and Gormley I. C. (2019) Investigation of parameter uncertainty in clustering using a Gaussian mixture model via jackknife, bootstrap and weighted likelihood bootstrap. <strong>Computational Statistics</strong>, 34, 1779—1813. <a href="https://doi.org/10.1007/s00180-019-00897-9" class="uri">https://doi.org/10.1007/s00180-019-00897-9</a></p>
<p>Fop M., Murphy T. B., Scrucca L. (2019) Model-based clustering with sparse covariance matrices. <strong>Statistics and Computing</strong>, 29:4, 791–819. <a href="https://doi.org/10.1007/s11222-018-9838-y" class="uri">https://doi.org/10.1007/s11222-018-9838-y</a></p>
<p>Scrucca L. (2019) A transformation-based approach to Gaussian mixture density estimation for bounded data. <strong>Biometrical Journal</strong>, 61:4, 873–888. <a href="https://doi.org/10.1002/bimj.201800174" class="uri">https://doi.org/10.1002/bimj.201800174</a></p>
<p>Ndabashinze B., Ustundag Siray G., Scrucca L. (2019) Genetic algorithms applied to fractional polynomials for power selection: application to diabetes data, <strong>Turkish Journal of Forecasting</strong>, 3, 15–25. <a href="https://doi.org/10.34110/forecasting.514761" class="uri">https://doi.org/10.34110/forecasting.514761</a></p>
<p>Giustozzi M., Vedovati M. C., Verso M., Scrucca L., Conti S., Verdecchia P., Bogliari G., Pierpaoli L., Agnelli G. and Becattini C. (2019) Patients aged 90 years or older with atrial fibrillation treated with oral anticoagulants: A multicentre observational study. <strong>International Journal of Cardiology</strong>, 281, 56–61. <a href="https://doi.org/10.1016/j.ijcard.2019.01.071" class="uri">https://doi.org/10.1016/j.ijcard.2019.01.071</a></p>
<p>Giustozzi M., Vedovati M. C., Verso M., Scrucca L., Conti S., Verdecchia P., Bogliari G., Pierpaoli L., Agnelli G. and Becattini C. (2019) Data on the use of oral anticoagulants in nonagenarians with atrial fibrillation, <strong>Data in Brief</strong>, Vol. 23, 103794. <a href="https://doi.org/10.1016/j.dib.2019.103794" class="uri">https://doi.org/10.1016/j.dib.2019.103794</a></p>
<p>Chirieleison C., Montrone A., Scrucca L. (2018) Eventi e mobilità sostenibile: un modello basato sulla cluster analysis. <strong>Economia della Cultura</strong>, anno XXVIII, 2018, n.&nbsp;4.</p>
<p>Scrucca L. and Raftery A. E. (2018) clustvarsel: A Package Implementing Variable Selection for Gaussian Model-based Clustering in R. <strong>Journal of Statistical Software</strong>, 84:1, 1–28. <a href="http://doi.org/10.18637/jss.v084.i01" class="uri">http://doi.org/10.18637/jss.v084.i01</a></p>
<p>Chirieleison C., Montrone A., Scrucca L. (2017) La mobilità sostenibile durante i grandi eventi italiani: un’analisi con il model-based clustering approach. <strong>Turistica</strong>, anno XXVI, vol.&nbsp;4, 35–55.</p>
<p>Chirieleison C. and Scrucca L. (2017) Event sustainability and transportation policy: a model-based cluster analysis for a cross-comparison of hallmark events. <strong>Tourism Management Perspectives</strong>, 24, 72–85. <a href="https://doi.org/10.1016/j.tmp.2017.07.020" class="uri">https://doi.org/10.1016/j.tmp.2017.07.020</a></p>
<p>Chirieleison C. and Scrucca L. (2017) Shaping students’ attitudes towards business ethics and corporate social responsibility: education versus personal perspectives, <strong>Education</strong>, 7:5, 83–95. <a href="http://article.sapub.org/10.5923.j.edu.20170705.01.html" class="uri">http://article.sapub.org/10.5923.j.edu.20170705.01.html</a></p>
<p>Scrucca L. (2017) On some extensions to GA package: hybrid optimisation, parallelisation and islands evolution, <strong>The R Journal</strong>, 9:1, 187–206. <a href="https://doi.org/10.32614/RJ-2017-008" class="uri">https://doi.org/10.32614/RJ-2017-008</a></p>
<p>Parisi A., Scrucca L., Desiderio J., Gemini A., Guarino S., Ricci F., Cirocchi R., Palazzini G., D’Andrea V., Minelli L., Trastulli S. (2017) Robotic right hemicolectomy: Analysis of 108 consecutive procedures and multidimensional assessment of the learning curve, <strong>Surgical Oncology</strong>, 26:1, 28–36. <a href="http://doi.org/10.1016/j.suronc.2016.12.005" class="uri">http://doi.org/10.1016/j.suronc.2016.12.005</a></p>
<p>Scrucca L., Fop M., Murphy T. B. and Raftery A. E. (2016) mclust 5: clustering, classification and density estimation using Gaussian finite mixture models, <strong>The R Journal</strong>, 8:1, 205–233. <a href="https://doi.org/10.32614/RJ-2016-021" class="uri">https://doi.org/10.32614/RJ-2016-021</a></p>
<p>Scrucca L. (2016) Identifying connected components in Gaussian finite mixture models for clustering. <strong>Computational Statistics &amp; Data Analysis</strong>. 93, 5–17. <a href="http://dx.doi.org/10.1016/j.csda.2015.01.006" class="uri">http://dx.doi.org/10.1016/j.csda.2015.01.006</a></p>
<p>Sahdra B., Ciarrochi J., Parker P. and Scrucca L. (2016) Using genetic algorithms in a large nationally representative American sample to abbreviate the Multidimensional Experiential Avoidance Questionnaire. <strong>Frontiers in Psychology</strong>, 7:189, 1–14. <a href="https://doi.org/10.3389/fpsyg.2016.00189" class="uri">https://doi.org/10.3389/fpsyg.2016.00189</a></p>
<p>Scrucca L. and Raftery A. E. (2015) Improved initialisation of model-based clustering using Gaussian hierarchical partitions. <strong>Advances in Data Analysis and Classification</strong>, 9:4, 447–460. <a href="https://doi.org/10.1007/s11634-015-0220-z" class="uri">https://doi.org/10.1007/s11634-015-0220-z</a></p>
<p>Cencetti C., de Rosa P., Fredduzzi A., Minelli A., and Scrucca L. (2015) A statistical test for drainage network recognition using MeanStreamDrop analysis. <strong>Geomatics, Natural Hazards and Risk</strong>, 5–7, 534–553. <a href="https://doi.org/10.1080/19475705.2014.897655" class="uri">https://doi.org/10.1080/19475705.2014.897655</a></p>
<p>Scrucca L. (2014) Graphical tools for model-based mixture discriminant analysis. <strong>Advances in Data Analysis and Classification</strong>, 8:2, 147–165. <a href="https://doi.org/10.1007/s11634-013-0147-1" class="uri">https://doi.org/10.1007/s11634-013-0147-1</a></p>
<p>Morris K., McNicholas P., and Scrucca L. (2013) Dimension reduction for model-based clustering via mixtures of multivariate t-distributions. <strong>Advances in Data Analysis and Classification</strong>, 7:3, 321–338. <a href="https://doi.org/10.1007/s11634-013-0137-3" class="uri">https://doi.org/10.1007/s11634-013-0137-3</a></p>
<p>Scrucca L. (2013) GA: A Package for Genetic Algorithms in R. <strong>Journal of Statistical Software</strong>, 5:4, 1–37. <a href="http://www.jstatsoft.org/v53/i04" class="uri">http://www.jstatsoft.org/v53/i04</a></p>
<p>Chirieleison C., Montrone A., Scrucca L. (2013) Measuring the impact of a profit oriented event on tourism: the Eurochocolate festival in Perugia, Italy. <strong>Tourism Economics</strong>, 19:6, 1411–1428. <a href="https://doi.org/10.5367%2Fte.2013.0269" class="uri">https://doi.org/10.5367%2Fte.2013.0269</a></p>
<p>Chirieleison C., Montrone A., Scrucca L. (2012) L’impatto degli eventi sul territorio: il caso Eurochocolate. <strong>AUR&amp;S</strong>, 7-8:12, 133–156.</p>
<p>Scrucca L. (2011) Model-based SIR for dimension reduction. <strong>Computational Statistics &amp; Data Analysis</strong>, 55:11, 3010–3026. <a href="https://doi.org/10.1016/j.csda.2011.05.006" class="uri">https://doi.org/10.1016/j.csda.2011.05.006</a></p>
<p>Scrucca L. (2010) Dimension Reduction for Model-Based Clustering. <strong>Statistics and Computing</strong>, 20:4, 471–484. <a href="https://doi.org/10.1007/s11222-009-9138-7" class="uri">https://doi.org/10.1007/s11222-009-9138-7</a></p>
<p>Scrucca L., Santucci A., Aversa F. (2010) Regression Modeling of Competing Risk Using R: An In Depth Guide for Clinicians. <strong>Bone Marrow Transplantation</strong>, 45, 1388–1395. <a href="https://doi.org/10.1038/bmt.2009.359" class="uri">https://doi.org/10.1038/bmt.2009.359</a></p>
<p>Scrucca L. (2007) Class prediction and gene selection for DNA microarrays using sliced inverse regression. <strong>Computational Statistics &amp; Data Analysis</strong>, 52, 438–451. <a href="https://doi.org/10.1016/j.csda.2007.02.005" class="uri">https://doi.org/10.1016/j.csda.2007.02.005</a></p>
<p>Scrucca L., Santucci A., Aversa F. (2007) Competing risks analysis using R: an easy guide for clinicians. <strong>Bone Marrow Transplantation</strong> (2007) 40, 381–387. <a href="https://doi.org/10.1038/sj.bmt.1705727" class="uri">https://doi.org/10.1038/sj.bmt.1705727</a></p>
<p>Scrucca L. (2004) L’identificazione dei distretti industriali e delle aree di sviluppo economico omogenee e/o integrate. <strong>Rivista dell’Agenzia Umbria Ricerche</strong>, 1, 71–90.</p>
<p>Scrucca L. (2004) qcc: an R package for quality control charting and statistical process control. <strong>R News</strong>, 4:1, June 2004, 11-17. <a href="https://cran.r-project.org/doc/Rnews/Rnews_2004-1.pdf" class="uri">https://cran.r-project.org/doc/Rnews/Rnews_2004-1.pdf</a></p>
<p>Scrucca L., Weisberg S. (2004) A simulation study to investigate the behavior of the log-density ratio under normality. <strong>Communications in Statistics (simulation and computation)</strong>, 33:1, 159–178. <a href="https://doi.org/10.1081/SAC-120028439" class="uri">https://doi.org/10.1081/SAC-120028439</a></p>
<p>De Angelis M., Scrucca L., Leandri M., Mincigrucci S., Bistoni S., Bovi M., Calabrese G., Pippi R., Parretti D., Grilli P., Colorio P., Fattorini M., Flamini O., Pacetti E., Travaglini A., Santeusanio F. (2003) Prevalence of carotid stenosis in type 2 diabetic patients asymptomatic for cerebrovascular disease. <strong>Diabetes, Nutrition &amp; Metabolism</strong>, 16:1, 48–55.</p>
<p>Scrucca L. (2002) Graphics for studying logistic regression models. <strong>Statistical Methods &amp; Applications</strong>, 11:3, 371–394. <a href="https://doi.org/10.1007/BF02509833" class="uri">https://doi.org/10.1007/BF02509833</a></p>
<p>Scrucca L. (2001) Nonparametric kernel smoothing methods. The sm library for Xlisp-Stat. <strong>Journal of Statistical Software</strong>, 6. <a href="http://www.jstatsoft.org/v06/i07" class="uri">http://www.jstatsoft.org/v06/i07</a></p>
<p>Scrucca L. (2001) A review and computer code for assessing the structural dimension of a regression model: uncorrelated 2D views. <strong>Computational Statistics &amp; Data Analysis</strong>, 36:2, 163–177. <a href="https://doi.org/10.1016/S0167-9473(00)00035-9" class="uri">https://doi.org/10.1016/S0167-9473(00)00035-9</a></p>
<p>Scrucca L. (2000) Assessing multivariate normality through interactive dynamic graphics. <strong>Quaderni di Statistica</strong>, 2, 221–240.</p>
<p>Scrucca (1996) I sondaggi elettorali nella recente esperienza italiana. <strong>Induzioni</strong>, n.&nbsp;12, 59–81.</p>
</div>
</section>
<section id="book-chapters" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="book-chapters">Book chapters</h2>
<div class="bibitem">
<p>Scrucca L. (2025) Capturing uncertainty in Gaussian mixture-based entropy estimation via weighted likelihood bootstrap. In <strong>Methodological and Applied Statistics and Demography IV</strong>, Pollice, A., Mariani, P. (editors). SIS 2024, Short Papers, Contributed Sessions 2. Springer, Cham. <a href="https://doi.org/10.1007/978-3-031-64447-4_91" class="uri">https://doi.org/10.1007/978-3-031-64447-4_91</a></p>
<p>Scrucca L., Saqr M., López-Pernas S., Murphy K. (2024) An introduction and R tutorial to model-based clustering in education via latent profile analysis. In <strong>Learning analytics methods and tutorials: A practical guide using R</strong>, Saqr M. &amp; López-Pernas S. (editors). Springer. <a href="https://lamethods.org/book1/chapters/ch09-model-based-clustering/ch9-model.html" class="uri">https://lamethods.org/book1/chapters/ch09-model-based-clustering/ch9-model.html</a></p>
<p>Serafini A., Scrucca L., Alfò M., Giordani P., Ferraro M. B. (2023) Fuzzy and model based clustering methods: can we fruitfully compare them? In: Brentari, E., Chiodi, M., Wit, EJ.C. (editors) <strong>Models for Data Analysis</strong>. SIS 2018, 49th Meeting of Italian Statistical Society, Palermo, Italy, June 20–22. Springer Proceedings in Mathematics &amp; Statistics, pp.&nbsp;283–304.</p>
<p>Scrucca L. (2016) Genetic algorithms for subset selection in model-based clustering. In <strong>Unsupervised Learning Algorithms</strong>, Celebi M. E. and Aydin K. (editors), Springer International Publishing, pp.&nbsp;55–70.</p>
<p>Chirieleison C., Scrucca L. (2016) CSR education in Italy. In <strong>Social Responsibility Education Across Europe. A Comparative Approach</strong>, Turker D., Altuntas Vural C., Idowu S. O. (editors), Springer International Publishing, pp.&nbsp;139–159.</p>
<p>Scrucca L., Bar-Hen A. (2013) A model-based dimension reduction approach to classification of gene expression data. In <strong>Advances in Theoretical and Applied Statistics</strong>, Torelli N., Pesarin F., Bar-Hen A. (editors), Springer, pp.&nbsp;221–230.</p>
<p>Scrucca L. (2011) A Geometric Approach to Subset Selection and Sparse Sufficient Dimension Reduction. In <strong>New Perspectives in Statistical Modeling and Data Analysis</strong>, editors Ingrassia S., Rocci R., Vichi M., Springer-Verlag, Berlin Heidelberg, pp.&nbsp;569–576.</p>
<p>Bartolucci F., Scrucca L. (2010) Point Estimation Methods with Applications to Item Response Theory Models. In <strong>International Encyclopedia of Education</strong>, Penelope Peterson, Eva Baker, Barry McGaw (editors), 3rd edition, Oxford: Elsevier, Vol. 7, pp.&nbsp;366–373.</p>
<p>Scrucca L. (2010) Visualization of model-based clustering structures. In <strong>Data Analysis and Classification</strong>, editors Palumbo F., Lauro C., Greenacre M., Berlin, Springer-Verlag, pp.&nbsp;67–75.</p>
<p>Scrucca L. (2006) Regularized sliced inverse regression with applications in classification. In <strong>Data Analysis, Classification and the Forward Search</strong>, editors Zani S., Cerioli A., Riani M., Vichi M., Berlin, Springer-Verlag, pp.&nbsp;59–66.</p>
<p>Scrucca L. (2005) La metodologia utilizzata per l’identificazione di aggregazioni sub-regionali omogenee. In <em>Verso una rilettura dei sistemi locali dell’Umbria: prime applicazioni di una nuova metodologia di clusterizzazione</em>, <strong>Quaderni PRASSI - Agenzia Umbria Ricerche</strong>, Cap. 2.</p>
<p>Scrucca L. (2003) Aree di sviluppo omogenee e/o integrate. Approcci e metodologie. <strong>Paper dell’Agenzia Umbria Ricerche</strong>, Collana: sviluppo e locale, n.&nbsp;8, pp.&nbsp;1–47.</p>
</div>
</section>
<section id="conferences-and-workshops" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="conferences-and-workshops">Conferences and workshops</h2>
<div class="bibitem">
<p>Scrucca L. (2024) A Model-Based Approach to Shot Charts Estimation in Basketball, <strong>Working Group on Model-Based Clustering</strong>. Bertinoro, July 22–26, 2024.</p>
<p>Scrucca, L. (2023) Gaussian Mixture Models for Probabilistic Learning, <strong>From Physics to Medicine: eXplainable AI workshop</strong>, Milano, November 20–22, 2023.</p>
<p>Scrucca, L. (2023) Anomaly Detection of Financial Ratios using Gaussian Mixture Models, <strong>Workshop Annuale del Dipartimento di Economia 2023 (WADE 2023)</strong>, Department of Economics, 30–31 October 2023 (Session II 31-10, Part 2.1.8), Zenodo. https://doi.org/10.5281/zenodo.10053896</p>
<p>Scrucca L. (2023) Mixture-based Estimation of Entropy and Its Applications, <strong>Working Group on Model-Based Clustering</strong>. Carnegie Mellon University, Pittsburgh (US), July 17–21, 2023.</p>
<p>Scrucca L. (2022) Gaussian mixtures using range-logit transformation with an application to image segmentation. <strong>MBC2 Book of Abstracts</strong>, 6th International Workshop on Models and Learning for Clustering and Classification, Catania, 31 August – 2 September 2022, pp.&nbsp;11–12.</p>
<p>Scrucca L. (2022) Silhouette Analysis for Model-Based Clustering, <strong>ECDA 2022, European Conference on Data Analysis</strong>, Naples, Italy, September 14–16 2022.</p>
<p>Scrucca L. (2021) Stacking ensemble of Gaussian mixtures for classification, <strong>Working Group on Model-Based Clustering</strong>. Athens University of Economics and Business, Athens, October 25–30, 2021.</p>
<p>Scrucca L. (2021) Stacking ensemble learning with Gaussian mixtures. In <strong>CLADAG 2021 Book of Abstracts and Short Papers. 13th Scientific Meeting of the Classification and Data Analysis Group</strong> (editors Giovanni C. Porzio, Carla Rampichini, Chiara Bocci), Firenze University Press, pp.&nbsp;420–423. Firenze, September 9–11 2021.</p>
<p>Scrucca L. (2021) Statistical models for infectious diseases epidemiology, <strong>COST Action: CA18208 - Novel tools for test evaluation and disease prevalence estimation</strong>, Training School, Perugia, September 16–17 2021.</p>
<p>Scrucca L., Serafini A. (2019) Mutual information Estimation via Gaussian Mixture Modelling, <strong>Working Group on Model-Based Clustering</strong>. WU Vienna University of Economics and Business, Vienna, July 14–20, 2019.</p>
<p>Scrucca L. (2019) A fast and efficient Modal EM algorithm for Gaussian mixtures. In <strong>CLADAG 2019 Book of Short Papers. 12th Scientific Meeting of the Classification and Data Analysis Group</strong> (editors Giovanni C. Porzio, Francesca Greselin, Simona Balzano), Edizioni Università di Cassino, pp.&nbsp;432–435. Cassino, September 11–13 2019.</p>
<p>Casa A., Scrucca L., Menardi G. (2018) Averaging via Stacking in Model-based Clustering, <strong>MBC2 – Workshop on Model Based Clustering and Classification</strong>. Catania, 5–7 September, 2018.</p>
<p>Alfò M., Ferraro M.B, Giordani P., Scrucca L., Serafini A. (2018) Model-based clustering vs Fuzzy clustering: a simulation comparison, <strong>Working Group on Model-Based Clustering</strong>. University of Michigan, Ann Arbor (USA), July 15–21, 2018.</p>
<p>Alfò M., Ferraro M.B, Giordani P., Scrucca L., Serafini A. (2018) A Comparison of Model-Based and Fuzzy Clustering Methods. <strong>SIS2018, 49th Scientific Meeting of the Italian Statistical Society</strong>. Palermo, 20–22 June, 2018. Abbruzzo A., Brentari E., Chiodi M., Piacentino D. (editors), Book of Short Papers SIS 2018, Pearson, pp.&nbsp;208–215.</p>
<p>Casa A., Scrucca L., Menardi G. (2018) On the Selection Uncertainty in Parametric Clustering, <strong>ECDA 2018, European Conference on Data Analysis</strong>, Paderborn, Germany, July 04–06 2018, Book of Abstracts, pp. 22–23.</p>
<p>Piersanti A., Tavoloni T., Scrucca L., Galarini R. (2018) Validation Data Processing by ADVeRSE 2.0 Free Software Package: Assessment of the Analytical Requirements of a GC-MS Method for the Determination of 16 PAHs in Smoked Fish, <strong>XXII International Mass Spectrometry Conference (IMSC)</strong>, 26–31 August 2018, Florence.</p>
<p>Fop M., Murphy T.B., Scrucca L. (2017) Model-based Clustering with Sparse Covariance Matrices. <strong>SIS 2017. Statistics and Data Science: new challenges, new generations</strong>. Florence, 28–30 June, 2017. Petrucci A., Verde R. (editors), Proceedings of the Conference of the Italian Statistical Society, pp.&nbsp;437–440.</p>
<p>Scrucca L., Fop M., Murphy T.B., Raftery A.E. (2017) The mclust R package for clustering, classification and density estimation using Gaussian finite mixture models, <strong>10th International Conference of the ERCIM WG on Computational and Methodological Statistics, 11th International Conference on Computational and Financial Econometrics</strong>, Senate House, University of London, UK, December 16-18 2017, Book of Abstracts, p.&nbsp;175.</p>
<p>Scrucca L. (2017) Boosting Gaussian mixture models for classification, <strong>Working Group on Model-Based Clustering</strong>. Università degli Studi di Perugia, Perugia, July 17—21, 2017.</p>
<p>Scrucca L. (2016) On some recent improvements to mixture modelling estimation: Gaussian hierarchical partitions based on data transformations and evolutionary EM, <strong>Working Group on Model-Based Clustering</strong>. Université Paris Descartes, Paris, July 17–23, 2016.</p>
<p>Scrucca L. (2016) Poisson change-point models estimated by genetic algorithms. In <strong>SIS2016, 48th Scientific Meeting of the Italian Statistical Society</strong>. Salerno, 8–10 June, 2016.</p>
<p>Scrucca L. (2015) A quick tour of new functionalities in MCLUST version 5, <strong>Working Group on Model-Based Clustering</strong>. University of Washington, Seattle, July 19–25, 2015.</p>
<p>Galarini R., Moretti S., Romanelli S., Scrucca L. (2015) Informatics tools to manage validation data of multiresidue analytical methods. MASSA 2015, Alghero (Italy), June 2015.</p>
<p>Scrucca L. (2014) Identifying connected components in Gaussian finite mixture models for clustering, <strong>Workshop on “Clustering methods and their applications”</strong>. Free University of Bozen - Bolzano, November 28th, 2014.</p>
<p>Scrucca L. (2014) Model-based density estimation for bounded data, <strong>Working Group on Model-Based Clustering</strong>. University College Dublin, Dublin, July 20–26, 2014.</p>
<p>Scrucca L., Raftery A. E. (2014) Improved initialisation of model-based clustering using a Gaussian hierarchical partition, <strong>MBC2 – Workshop on Model Based Clustering and Classification</strong>. Catania, 3–5 September, 2014.</p>
<p>Scrucca L. (2013) Identifying connected components in model-based clustering, <strong>Working Group on Model-Based Clustering</strong>. Università di Bologna, July 21–27, 2013.</p>
<p>Scrucca L. (2013) On the implementation of a parallel algorithm for variable selection in model-based clustering. <strong>CLADAG 2013. 9th Scientific Meeting of the Classification and Data Analysis Group. Book of Abstracts</strong> (editors Tommaso Minerva, Isabella Morlini, Francesco Palumbo). Cleup. 9th Meeting of the CLassification and Data Analysis Group of the Italian Statistical Society, Modena, 18–20 September 2013.</p>
<p>Scrucca L. (2012) MCLUST version 4 session, <strong>Working Group on Model-Based Clustering</strong>. University of Guelph, July 16–20, 2012.</p>
<p>Scrucca L. (2012) Graphical tools for model-based mixture discriminant analysis, <strong>MBC2 – Workshop on Model Based Clustering and Classification</strong>, Catania, 6–7 September 2012</p>
<p>Scrucca L. (2011) Genetic algorithms for subset selection in model-based clustering, <strong>Working Group on Model-Based Clustering</strong>, University of Glasgow, Glasgow, July 17–23, 2011.</p>
<p>Scrucca L. (2011) Classification on a dimension reduced subspace. <strong>CLADAG 2011, Book of Abstracts</strong> (editors Paola Cerchiello, Claudia Tarantola), Pavia University Press, p.&nbsp;47. 8th Scientific Meeting of the CLAssification and Data Analysis Group of the Italian Statistical Society, Pavia, 7–9 September 2011.</p>
<p>Scrucca L. (2010) Some recent advances on dimension reduction for high-dimensional data. In <strong>GfKL-CLADAG 2010, Book of Abstracts</strong>, pp. 99–100. Joint meeting GfKl-CLADAG 2010, 8–10 September 2010.</p>
<p>Scrucca L., Bar-Hen A. (2010) A model-based dimension reduction approach to classification of gene expression data. In <strong>SIS2010, Atti della XLV Riunione Scientifica della Società Italiana di Statistica</strong>, CLEUP. Padova, 16–18 June, 2010.</p>
<p>Scrucca L. (2010) Genetic algorithms for subset selection in model-based clustering. In <strong>SIS2010, Atti della XLV Riunione Scientifica della Società Italiana di Statistica</strong> CLEUP. Padova, 16-18 June, 2010.</p>
<p>G. Saluti, Galarini R., Scrucca L. (2010) Multivariate Charts for Quality Control of Multiresidue Analytical Methods. In <em>Sixth International Symposium on Hormone and Veterinary Drug Residue Analysis</em>, Ghent (Belgium) 1-4 June 2010.</p>
<p>Scrucca L. (2009) A Finite Mixture Model Approach to Sliced Inverse Regression, <strong>Working Group on Model-Based Clustering</strong>. Institut Henri Poincaré, Paris, July 19–25, 2009.</p>
<p>Scrucca L. (2009) A Geometric Approach to Subset Selection and Sparse Sufficient Dimension Reduction. In <strong>CLADAG 2009, Book of Short Papers</strong> (editors Ingrassia S. e Rocci R.), Padova, CLEUP, pp.&nbsp;235–238. VII Riunione Scientifica del Gruppo di Classificazione e Analisi dei Dati della Società Italiana di Statistica, Catania, 9–11 September 2009.</p>
<p>Scrucca L. (2008) A mixture model approach to dimension reduction in regression (Un approccio basato sui modelli mistura per la riduzione della dimensionalità nell’ambito della regressione). In <strong>SIS2008, Atti della XLIV Riunione Scientifica della Società Italiana di Statistica</strong>, pp.&nbsp;233–240. Arcavacata di Rende, 25–27 June, 2008.</p>
<p>Scrucca L. (2007) Dimension reduction and visualization for model-based clustering, <strong>Working Group on Model-Based Clustering</strong>. Trinity College Dublin, Dublin, July 16–20, 2007.</p>
<p>Scrucca L. (2007) Visualization of model-based clustering structures. In <strong>CLADAG 2007, Book of Short papers</strong>, Lauro C. e Palumbo F. (editors), pp.&nbsp;451–454. VI Riunione Scientifica del Gruppo di Classificazione e Analisi dei Dati della Società Italiana di Statistica, Macerata 12-14 September 2007.</p>
<p>Scrucca L. (2006) Dimension reduction and visualization for model-based clustering. In <strong>COMPSTAT 2006, Book of abstracts</strong>, 17th Symposium of IASC on Computational Statistics, Rome, 28 Agust - 1 September 2006.</p>
<p>Piersanti A., Galarini R., Tavoloni T., Scrucca L. (2006) Polychlorobiphenyl (18 congeners) in mussels from middle adriatic sea. In <strong>Proceedings of 26th International Symposium on Halogenated Persistent Organic Pollutants - DIOXIN 2006</strong>, Oslo, 21-25 August 2006, pp.&nbsp;1951–1954. Organohalogen Compounds Vol. 68.</p>
<p>Scrucca L. (2005) Regularized sliced inverse regression with applications in classification. In <strong>CLADAG 2005, Book of Short papers</strong> (editors Zani S. e Cerioli A.), pp.&nbsp;509–512. V Riunione Scientifica del Gruppo di Classificazione e Analisi dei Dati della Società Italiana di Statistica, Parma 6-8 giugno 2005.</p>
<p>Scrucca L. (2004) Class prediction and gene selection for DNA microarrays using sliced inverse regression. In <strong>Proceedings of the 19th International Workshop on Statistical Modelling (IWSM 2004)</strong>, Firenze, 4-8 July 2004, pp.&nbsp;259–263.</p>
<p>Scrucca L. (2003) On a test procedure for assessing a parametric approximation to a nonlinear term in semiparametric models. In <strong>Proceedings ISI 2003 - International Statistical Institute</strong> (13-20 August 2003, Berlin, Germany), Contributed Papers, Vol. LX, Book 2, pp. 397–398.</p>
<p>Scrucca L. (2002) Robust sliced inverse regression and outliers detection. In <strong>Euroworkshop on Statistical Modelling Model Building and Evaluation</strong>, Bernried, Germany, Oct 31-Nov 3 2002.</p>
<p>Porzio G.C., Scrucca L. (2002) Visualizing the correlation coefficient (Il coefficiente di correlazione: una sua rappresentazione grafica). In <strong>SIS2002, Atti della XLI Riunione Scientifica della Società Italiana di Statistica</strong>, Milano, 5-7 Giugno, 2002.</p>
<p>Attygalle D., Francis B., Scrucca L. (2002) Model coverage, forward searching and multiple outlier detection. In <strong>17th International Workshop on Statistical Modelling</strong>. Crete, Greece, 8-12 July, 2002.</p>
<p>Ranalli M.G., Scrucca L. (2002) Understanding influential observations in regression estimation of finite population means using graphs. In <strong>COMPSTAT 2002, Book of abstracts</strong>, 15th Symposium of IASC on Computational Statistics, Berlin, Germany, 24-28 August 2002.</p>
<p>Scrucca L. (2001) Visualizing and assessing partial correlation coefficients through added variable plots. In <strong>Atti del Convegno S.Co. 2001, Modelli Complessi e Metodi Computazionali Intensivi per la Stima e la Previsione</strong>, Bressanone, 24-26 Settembre 2001, a cura di C. Provasi, Cleup Editrice, Padova.</p>
<p>Scrucca L. (1999) Dynamic probability plot for assessing multivariate normality. In <strong>Atti del Convegno S.Co. 1999, Modelli Complessi e Metodi Computazionali Intensivi per la Stima e la Previsione</strong>, Venezia, 27-29 Settembre 1999, pp.&nbsp;122–129.</p>
</div>
</section>
<section id="reports" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="reports">Reports</h2>
<div class="bibitem">
<p>Scrucca L., Raftery A. E. (2014) clustvarsel: A Package Implementing Variable Selection for Model-based Clustering in R. <strong>Technical Report, Department of Statistics, University of Washington</strong>, No.&nbsp;629, pp. 1–27.</p>
<p>Fraley C., Raftery A. E., Murphy T. B., Scrucca L. (2012) mclust version 4 for R: Normal Mixture Modeling for Model-Based Clustering, Classification, and Density Estimation. <strong>Technical Report, Department of Statistics, University of Washington</strong>, No.&nbsp;597, pp.&nbsp;1–58</p>
<p>Scrucca L. (2006) Subset selection in dimension reduction methods. <strong>Quaderni del Dipartimento di Economia, Finanza e Statistica</strong>, Università degli Studi di Perugia, n.&nbsp;23, Settembre 2006.</p>
<p>Scrucca L. (2005) Clustering multivariate spatial data based on local measures of spatial autocorrelation. An application to the labour market of Umbria. <strong>Quaderni del Dipartimento di Economia, Finanza e Statistica</strong>, Università degli Studi di Perugia, n.&nbsp;20, Ottobre 2005.</p>
</div>
</section>
<section id="miscellanea" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="miscellanea">Miscellanea</h2>
<div class="bibitem">
<p>Scrucca L. (2022) Pandemia e previsioni: per l’Umbria una curva in discesa. <strong>AUR Focus</strong>.<br>
<a href="https://www.agenziaumbriaricerche.it/focus/pandemia-e-previsioni-per-lumbria-una-curva-in-discesa/" class="uri">https://www.agenziaumbriaricerche.it/focus/pandemia-e-previsioni-per-lumbria-una-curva-in-discesa/</a></p>
<p>Scrucca L. (2021) Incidenza e impatto sul settore ospedaliero della pandemia CoViD-19: un confronto tra regioni. <strong>AUR Focus</strong>.<br>
<a href="https://www.agenziaumbriaricerche.it/focus/incidenza-e-impatto-sul-settore-ospedaliero-della-pandemia-covid-19-un-confronto-tra-regioni/" class="uri">https://www.agenziaumbriaricerche.it/focus/incidenza-e-impatto-sul-settore-ospedaliero-della-pandemia-covid-19-un-confronto-tra-regioni/</a></p>
<p>Serafini A., Murphy T. B., Scrucca L. (2020) Handling missing data in model-based clustering. <strong>arXiv:2006.02954</strong> pre-print available at <a href="https://arxiv.org/abs/2006.02954" class="uri">https://arxiv.org/abs/2006.02954</a></p>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/luca-scr\.github\.io");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2024 Luca Scrucca</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>Website created with <a href="https://quarto.org/">Quarto</a></p>
</div>
  </div>
</footer>




</body></html>